src/shims/tls.rs

```diff
@@ -33,24 +38,23 @@ pub struct TlsData<'tcx> {
     /// pthreads-style thread-local storage.
     keys: BTreeMap<TlsKey, TlsEntry<'tcx>>,
 
-    /// A single global per thread dtor (that's how things work on macOS) with a data argument.
-    global_dtors: BTreeMap<ThreadId, (ty::Instance<'tcx>, Scalar<Tag>)>,
-
-    /// Whether we are in the "destruct" phase, during which some operations are UB.
-    dtors_running: HashSet<ThreadId>,
+    /// A single per thread destructor of the thread local storage (that's how
+    /// things work on macOS) with a data argument.
+    thread_dtors: BTreeMap<ThreadId, (ty::Instance<'tcx>, Scalar<Tag>)>,
```

Comment:
Either this should also mention `macos` in the field name of some of the methods shouldn't... right now that's a bit inconsistent.